src/features/query-bar/components/filter/Condition.js

```diff
@@ -138,8 +163,7 @@ const Condition = ({
             default:
                 break;
         }
-
-        const error = areErrorsEnabled && !isValueSet ? message : null;
+        const error = isValueInvalid || (areErrorsEnabled && !isValueSet) ? message : null;
```

Comment:
I think you should return null in the beginning if there are no errors.

Reply:
Do you mean to do:
```
const error = isValueInvalid || (areErrorsEnabled && !isValueSet) ? message : null;
        if (!error) {
            return null;
        }
        return error;
```

Reply:
I mean you can have something like this in line 136:
`      if (!isValueInvalid && (!areErrorsEnabled || isValueSet)) { return null; }`

This way, you don't have to do the switch statement if there are no errors.

Come to think of it, shouldn't you just return `null` if `areErrorsEnabled` is `false`? haha
Also, that should probably have been named `isErrorEnabled`.
We can take some time to talk through the logic here